src/libexpr/primops/fetchTree.cc

```diff
@@ -438,6 +444,30 @@ static RegisterPrimOp primop_fetchGit({
         With this argument being true, it's possible to load a `rev` from *any* `ref`
         (by default only `rev`s from the specified `ref` are supported).
 
+      - `verifyCommit` (default: `true` if `publicKey` or `publicKeys` are provided, otherwise `false`)
+
+        Whether to check `rev` for a signature matching publicKey or publicKeys.
+        If `verifyCommit` is enabled, then `fetchGit` cannot use a repository with
+        uncommitted changes.
+
+      - `publicKey`
+
+        The public key against which `rev` is verified if `verifyCommit` is enabled.
+
+      - `keytype` (default: `ssh-ed25519`)
+
+        The keytype of `publicKey`.
+        Possible values are `ssh-dsa`, `ssh-ecdsa`, `ssh-ecdsa-sk`, `ssh-ed25519`,
+        `ssh-ed25519-sk` or `ssh-rsa`.
+
+      - `publicKeys`
+
+        The public key against which `rev` is verified if `verifyCommit` is enabled.
+        Must be given as a list of attribute sets, all of which contain the attribute
+        `key`, the public key, and optionally the attribute `type`, the keytype
+        (default: `ssh-ed25519`).
```

Comment:
````suggestion
        Must be given as a list of attribute sets with the following form:
        ```nix
        {
          key = "<public key>";
          type = "<key type>"; # optional, default: "ssh-ed25519"
        }
        ```
````